internal/cli/atlas/deployments/setup.go

```diff
@@ -0,0 +1,238 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deployments
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/mongosh"
+	"github.com/mongodb/mongodb-atlas-cli/internal/podman"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+const (
+	startHostPort          = 37017
+	internalMongodPort     = 27017
+	internalMongotPort     = 27027
+	localCluster           = "LOCAL"
+	mdb6                   = "6.0"
+	InternalMongodHostname = "mongod"
+	InternalMongotHostname = "mongot"
+	ReplicaSetName         = "rs-localdev"
+)
+
+type SetupOpts struct {
+	cli.OutputOpts
+	cli.GlobalOpts
+	deploymentName string
+	deploymentType string
+	mdbVersion     string
+	port           int
+	debug          bool
+}
+
+const startTemplate = `local environment started at {{.ConnectionString}}
+`
+
+func (opts *SetupOpts) createLocalDeployment() error {
```

Comment:
I suggest breaking down this method a bit more, like `validateDeployment`, `createMongoD` and `createMongoT`

Reply:
done